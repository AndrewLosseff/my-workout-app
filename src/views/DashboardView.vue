<script setup lang="ts">
import { onMounted, computed } from 'vue'
import { useWorkoutStore } from '@/stores/workout'
import { useExerciseStore } from '@/stores/exercise'
import WorkoutCard from '../components/WorkoutCard/index.vue'

const ZERO_WORKOUTS = 0

const workoutStore = useWorkoutStore()
const exerciseStore = useExerciseStore()

onMounted(() => {
  workoutStore.loadFromLocalStorage()
  exerciseStore.loadFromLocalStorage()
})

const workouts = computed(() => workoutStore.getWorkouts)

const deleteWorkout = (id: string) => {
  workoutStore.deleteWorkout(id)
}
</script>

<template>
  <div>
    <h1 class="text-h4 mb-4">Your Workouts</h1>

    <v-alert
      v-if="workouts.length === ZERO_WORKOUTS"
      class="mb-4"
      type="info"
      text="You don't have any workouts yet. Add your first workout to get started!"
    ></v-alert>

    <v-row>
      <v-col v-for="workout in workouts" :key="workout.id" cols="12" sm="6" md="4">
        <workout-card :workout="workout" @delete="deleteWorkout" />
      </v-col>

      <v-col cols="12" sm="6" md="4">
        <v-card elevation="2" class="custom-card">
          <v-card-title> Add New Workout </v-card-title>
          <div class="button-container">
            <v-btn color="primary" fab to="/add-workout">
              <v-icon>mdi-plus</v-icon>
            </v-btn>
          </div>
        </v-card>
      </v-col>
    </v-row>
  </div>
</template>

<style scoped>
.custom-card {
  height: 100%;
  min-height: 210px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.button-container {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-grow: 1;
}
</style>
