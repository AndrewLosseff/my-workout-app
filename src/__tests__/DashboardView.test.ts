import { describe, it, expect, beforeEach, vi } from 'vitest'
import { mount, VueWrapper } from '@vue/test-utils'
import { createPinia, setActivePinia } from 'pinia'
import DashboardView from '@/views/DashboardView.vue'

vi.mock('@/stores/workout', () => ({
  useWorkoutStore: () => ({
    loadFromLocalStorage: vi.fn(),
    getWorkouts: [],
    deleteWorkout: vi.fn()
  })
}))

vi.mock('@/stores/exercise', () => ({
  useExerciseStore: () => ({
    loadFromLocalStorage: vi.fn()
  })
}))

describe('DashboardView', () => {
  let wrapper: VueWrapper

  beforeEach(() => {
    setActivePinia(createPinia())

    wrapper = mount(DashboardView, {
      global: {
        stubs: {
          'v-btn': true,
          'v-icon': true,
          'v-card': true,
          'v-card-title': true,
          'v-alert': true,
          'v-row': true,
          'v-col': true,
          'workout-card': true
        }
      }
    })
  })

  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('shows empty state when no workouts exist', () => {
    expect(wrapper.find('v-alert-stub').exists()).toBe(true)
    expect(wrapper.find('h1').text()).toBe('Your Workouts')
  })
})
