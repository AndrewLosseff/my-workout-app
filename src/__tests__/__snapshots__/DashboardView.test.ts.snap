// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DashboardView > renders correctly 1`] = `
"<div data-v-336c5134="">
  <h1 data-v-336c5134="" class="text-h4 mb-4">Your Workouts</h1>
  <v-alert-stub data-v-336c5134="" class="mb-4" type="info" text="You don't have any workouts yet. Add your first workout to get started!"></v-alert-stub>
  <v-row-stub data-v-336c5134=""></v-row-stub>
</div>"
`;
